pkg/apis/eksctl.io/v1alpha5/validation.go

```diff
@@ -224,3 +228,27 @@ func countEnabledFields(fields ...*string) int {
 	}
 	return count
 }
+
+func validateNodeGroupKubeletExtraConfig(kubeletConfig *NodeGroupKubeletConfig) error {
+	if kubeletConfig == nil {
+		return nil
+	}
+
+	var kubeletForbiddenFields = map[string]struct{}{
+		"kind":               {},
+		"apiVersion":         {},
+		"address":            {},
+		"clusterDomain":      {},
+		"authentication":     {},
+		"authorization":      {},
+		"serverTLSBootstrap": {},
+		"featureGates":       {},
```

Comment:
Actually, we may wish to allow appending feature gates, but let's wait until use-cases emerge :)